src/internal/operators/connect.ts

```diff
@@ -0,0 +1,44 @@
+/** @prettier */
+import { lift } from '../util/lift';
+import { Subscriber } from '../Subscriber';
+import { OperatorFunction, ObservableInput } from '../types';
+import { Observable } from '../Observable';
+import { Subject } from '../Subject';
+import { from } from '../observable/from';
+import { Subscription } from '../Subscription';
+
+function defaultConnector<T>() {
+  return new Subject<T>();
+}
+export function connect<T, R>({
+  connector = defaultConnector,
+  setup,
+}: {
+  connector?: () => Subject<T>;
+  setup: (shared: Observable<T>) => ObservableInput<R>;
```

Comment:
nitpick: I'm not a fan of the `setup` property's name. TBH, I preferred `selector`. My main beef with `setup` is that, IMO, it doesn't really explain what the parameter does. [Elsewhere in the API](https://github.com/ReactiveX/rxjs/blob/8ef4fee2b5f7fb68426661ffcc47b50ab81b145e/src/internal/operators/catchError.ts#L12) functions that are used in this manner are called selectors and I am a fan of consistency and precedent.

Reply:
how about `project`?

Reply:
`selector`, IMO, also doesn't explain what the parameter does, either. I mean, this sets up the multicast as much as it selects the output. Happy to bikeshed the name, but overall, I guess I'm just not sure what to name this. `project` is used elsewhere to mean a synchronous map-type function.

Reply:
Maybe `cast`? or `multicast`? or??

Reply:
(I'm open to moving to back to the `selector`, though). I just don't like that it doesn't completely imply what's going on here. `catchError` is a little different, because it's only called when an error happens, and you have to "select" what to observe after that error shuts things down.

This function is required to setup the multicast. It also selects what sort of values are emitted from the result. I guess I'm now unhappy with either `setup` or `select` haha. But I do know I want it to be terse.

Reply:
> This function is required to setup the multicast. It also selects what sort of values are emitted from the result.

Well, the function is "required to setup" the multicast, but it doesn't actually set it up. It receives the shared/multicast observable and then "selects what sort of values are emitted from the result".

That said, `cast` or `multicast` are okay, I suppose, but my preference is still for `selector`.